src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -73,22 +72,49 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   // Send a user to datastore. Update the current information about the user if one exists.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGitHubID = request.getParameter("githubID");
-    if (userGitHubID == null || userGitHubID.trim().length() == 0) {
+    String userGitHubAuthToken = request.getParameter("gitHubAuthToken");
+    if (userGitHubAuthToken == null) {
+      ErrorResponse.sendJsonError(
+          response,
+          "GitHub OAuth token was never supplied",
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "User could not be authenticated by GitHub, please try again");
+      return;
+    }
+    String[] interestTagsParam = request.getParameterValues("interestTags");
+    if (interestTagsParam == null) {
+      interestTagsParam = new String[] {};
+    }
+    ArrayList<String> userTags = new ArrayList<>(Arrays.asList(interestTagsParam));
+
+    User user = AuthServlet.getAuthorizedUser();
+    if (user == null) {
       ErrorResponse.sendJsonError(
           response,
-          "Empty GitHub ID",
-          HttpServletResponse.SC_BAD_REQUEST,
-          "There is no GitHub ID affiliated with this request");
+          "User not logged in",
+          HttpServletResponse.SC_FORBIDDEN,
```

Comment:
nit: Could we change this to 401 instead of 403? Forbidden (403) is more often used to indicate that a user is not permitted to access something, where unauthorized (401) is more often used to indicate that a user must authenticate and try again. [See here](https://stackoverflow.com/a/28672217)
```suggestion
          HttpServletResponse.SC_UNAUTHORIZED,
```

Reply:
That makes sense! these errors are so confusing haha